pkg/streamingpromql/operators/series_merging_test.go

```diff
@@ -0,0 +1,763 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package operators
+
+import (
+	"testing"
+
+	"github.com/prometheus/prometheus/model/histogram"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/limiting"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+func TestMergeSeries(t *testing.T) {
+	testCases := map[string]struct {
+		input               []types.InstantVectorSeriesData
+		sourceSeriesIndices []int
```

Comment:
(nit) would be good to document what these variables represent (or at least sourceSeriesIndices in the context of the test case)

Reply:
`sourceSeriesIndices` has the same meaning as the parameter to `MergeSeries`, and this is already documented for `MergeSeries` - is there something special you think we should mention in the context of the tests?

Reply:
That's fair enough, although I don't think it would hurt to have it in the tests for clarity on how to write tests, but I'm not worried about it.